src/annotation.js

```diff
@@ -26,6 +26,19 @@ export default {
 
 	beforeUpdate(chart, args, options) {
 		const annotationOptions = options || args;
+
+		if (isObject(annotationOptions.annotations)) {
+			const array = new Array();
+			Object.keys(annotationOptions.annotations).forEach(key => {
+				let value = annotationOptions.annotations[key];
+				if (isObject(value)) {
+					value.id = key;
+					array.push(value);
+				}
+			});
+			annotationOptions.annotations = array;
```

Comment:
This modification here requires that the provided options object is not the one that user provided.
I think its currently that way and will probably stay like that (because its merged with defaults).

BTW, the options || args can be removed, since beta.5 of chart.js is out?

Reply:
> BTW, the options || args can be removed, since beta.5 of chart.js is out?

Yes, I think it could be removed. Nevertheless I'm not familiar with npm packages... To set BETA5 I think it could be easy, I'm gonna change the files, but how to update the node modules? just `npm update`?

Reply:
That should do it. But no need to do that in this PR.

Reply:
> 
> 
> That should do it. But no need to do that in this PR.

OK! I don't do it